src/core/datastore/networkstore.js

```diff
@@ -88,42 +97,15 @@ export class NetworkStore {
    * @return  {Observable}                                                Observable.
    */
   find(query, options = {}) {
-    const useDeltaFetch = options.useDeltaFetch === true || this.useDeltaFetch;
-    const stream = KinveyObservable.create((observer) => {
-      // Check that the query is valid
-      if (isDefined(query) && !(query instanceof Query)) {
-        return observer.error(new KinveyError('Invalid query. It must be an instance of the Query class.'));
-      }
-
-      // Create the request
-      const config = {
-        method: RequestMethod.GET,
-        authType: AuthType.Default,
-        url: url.format({
-          protocol: this.client.apiProtocol,
-          host: this.client.apiHost,
-          pathname: this.pathname
-        }),
-        properties: options.properties,
-        query: query,
-        timeout: options.timeout,
-        client: this.client
-      };
-      let request = new KinveyRequest(config);
-
-      // Should we use delta fetch?
-      if (useDeltaFetch === true) {
-        request = new DeltaFetchRequest(config);
-      }
+    const errPromise = this._ensureValidQuery(query);
```

Comment:
Okay, I can name it `_validateQuery()`. I usually call methods `ensure...` when they throw exceptions, but I guess this "evolved" and I forgot to change it at some point.

The goal is to avoid duplication, yes, as it's done in a few methods of the stores. It would be nice to have a central place to create errors, but there isn't such a system right now and it felt out of scope. I'd rather we leave it like that for the redesign, but we do have to start somewhere and if you think it's best to do it now, we can.

Reply:
Doesn't have to happen now. Just one of those things that, if we think is a good idea, should be logged for the future.

Reply:
Here's a ticket for it - [MLIBZ-2294](https://kinvey.atlassian.net/browse/MLIBZ-2294).